pkg/autoscaler/statforwarder/processor.go

```diff
@@ -54,11 +74,49 @@ func (p *bucketProcessor) process(sm asmetrics.StatMessage) error {
 		return err
 	}
 
-	return p.conn.SendRaw(gorillawebsocket.BinaryMessage, b)
+	if p.podAddressable && p.podConn.SendRaw(gorillawebsocket.BinaryMessage, b) == nil {
+		// Pod is accessible via IP address, close the connection via SVC.
+		if p.svcConn != nil {
+			if err := p.svcConn.Shutdown(); err != nil {
+				p.logger.Warnw("Failed to close connection", zap.Error(err))
+			} else {
+				p.svcConn = nil
+			}
+		}
+		return nil
+	}
+
+	if p.svcConn == nil {
+		p.logger.Info("Connecting to Autoscaler bucket at ", p.svcDNS)
+		p.svcConn = websocket.NewDurableSendingConnection(p.svcDNS, p.logger)
+	}
+
+	if err := p.svcConn.SendRaw(gorillawebsocket.BinaryMessage, b); err != nil {
+		// Sending via IP address and SVC both fail, return the error.
+		return err
+	}
+
+	// Pod is accessible via SVC only, mark podAddressable false and close the connection via Pod IP.
+	if p.podAddressable {
+		p.logger.Info("Autoscaler pods can't be accessed by IP address")
+		p.podAddressable = false
+	}
+	if p.podConn != nil {
```

Comment:
ditto, this can be only if podAdressable is still true so no need  to do this everytime.